app-website/src/pages/games/[...slug].astro

```diff
@@ -0,0 +1,18 @@
+---
+import { getCollection } from 'astro:content';
+import GameLayout from '../../layouts/GameLayout.astro';
+
+export async function getStaticPaths() {
+  const gamesEntries = await getCollection('games');
+  return gamesEntries.map(entry => ({
+    params: { slug: entry.slug }, props: { entry },
+  }));
+}
```

Comment:
Could we move all of the imperative code to `.ts` files (service layer) and keep the `.astro` files declarative as a convention?